File: app-core/src/main/java/ru/micron/persistence/model/FilmRating.java
package ru.micron.persistence.model;

import java.util.Objects;
import java.util.UUID;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.Hibernate;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "film_rating")
public class FilmRating {

  @Id
  @Column(name = "uuid")
  private UUID uuid;

  @Column(name = "imdb_id")
  private String imdbId;

  @Column(name = "source")
  private String source;

  @Column(name = "value")
  private String value;

  @ManyToOne(cascade = CascadeType.ALL)
  @JoinColumn(name = "imdb_id", insertable = false, updatable = false)
  private Film film;

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) {
      return false;
    }
    FilmRating that = (FilmRating) o;
    return uuid != null && Objects.equals(uuid, that.uuid);
  }

  @Override
  public int hashCode() {
    return getClass().hashCode();
  }
}

File: app-core/src/main/kotlin/ru/micron/persistence/model/FilmRating.kt
package ru.micron.persistence.model

import org.hibernate.Hibernate
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "film_rating")
data class FilmRating(
    @Id
    @Column(name = "uuid", nullable = false)
    var uuid: UUID? = null,

    @Column(name = "imdb_id")
    var imdbId: String? = null,

    @Column(name = "source")
    var source: String? = null,

    @Column(name = "value")
    var value: String? = null,

    @ManyToOne(fetch = FetchType.EAGER, cascade = [CascadeType.ALL])
    @JoinColumn(name = "imdb_id", insertable = false, updatable = false)
    val film: Film? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) {
            return false
        }
        val that = other as FilmRating
        return uuid != null && uuid == that.uuid
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}